### PR TITLE
Decode query result

### PR DESCRIPTION
**Pull request by LudovicoGranata**
- Code to decode query result
- Update of the buyQuery function to decode data
- Update of the test function to retrieve the final result

### CODE REVIEW

File: src/test/desmo-test.ts [hunk 1]
```diff
@@ -56,14 +55,17 @@ describe('Desmo Tests', function () {
         property: { identifier: 'value', unit: 'qudt:DEG_C', datatype: 1 },
         staticFilter: '$[?(@["type"]=="Sensor")]',
       };
+      */
```
Comment (relu91) [inline src/test/desmo-test.ts:58]: Remove commented code

**Review by relu91 — DISMISSED** [review]
Add some unit tests for decoder.ts. I expect one for each type and also some failing test for malformed payloads.

File: src/test/desmo-test.ts (hunk 1 above)
Comment (iosonopersia) [reply src/test/desmo-test.ts]: Done.

**Review by relu91 — APPROVED** [review]
LGTM. Merging...

**Pull request merged**